### PR TITLE
GitHub service connection API breaking change

### PR DESCRIPTION
**Pull request by xuzhang3**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Issue Number:

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?
#72 

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go [hunk 1]
```diff
@@ -4,12 +4,132 @@
...
+					resource.TestCheckResourceAttr(tfSvcEpNode, "service_endpoint_name", serviceEndpointName),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "description", "Managed by Terraform"),
+					testutils.CheckServiceEndpointExistsWithName(tfSvcEpNode, serviceEndpointName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccServiceEndpointGitHub_PersonalTokenUpdate(t *testing.T) {
```
Comment (EliiseS) [inline azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:39]: Doesn't this test cover the `TestAccServiceEndpointGitHub_PersonalTokenBasic`? Do you need to have both?

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go [hunk 2]
```diff
@@ -4,12 +4,132 @@
...
+					resource.TestCheckResourceAttrSet(tfSvcEpNode, "project_id"),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "auth_oauth.#", "1"),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "service_endpoint_name", serviceEndpointName),
+					testutils.CheckServiceEndpointExistsWithName(tfSvcEpNode, serviceEndpointName),
+				),
+			},
+		},
+	})
+}
+func TestAccServiceEndpointGitHub_OauthUpdate(t *testing.T) {
```
Comment (EliiseS) [inline azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:97]: Isn't the test `TestAccServiceEndpointGitHub_OauthBasic` covered in this test?

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go [hunk 3]
```diff
@@ -53,3 +173,63 @@ func TestAccServiceEndpointGitHub_CreateAndUpdate(t *testing.T) {
...
+	service_endpoint_name  = "%[1]s"
+	auth_oauth {
+		oauth_configuration_id= "xxxx-xxxx-xxxx-xxxx-xxxx"
+	}
+}`, serviceEndpointName)
+
+	return fmt.Sprintf("%s\n%s", projectResource, serviceEndpointResource)
+}
+
+func oauthConfigUpdate(projectName string, serviceEndpointName string, description string) string {
```
Comment (EliiseS) [inline azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:222]: Do we not want to put them into the hcl.go anymore?

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go (hunk 2 above)
Comment (xuzhang3) [reply azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:97]: Yes, `basic` is used to create the base service connection then `update` can update the service connection and verify it

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go (hunk 3 above)
Comment (xuzhang3) [reply azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:222]: I plan to refactor the code
1. The `HCL` test configure is used only in`resource_serviceendpoint_github_test.go`,  no need to make this test data public
2. All test `HCL` in one file, soon we will get a monster file, keep the test data where is belongs will be more clear.

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go (hunk 1 above)
Comment (xuzhang3) [reply azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:39]: Yes, `basic ` is used to create the base service connection then `update ` will update the service connection just created and verify it

File: azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go (hunk 3 above)
Comment (tmeckel) [reply azuredevops/internal/acceptancetests/resource_serviceendpoint_github_test.go:222]: I agree but you still need some place where you put shared HCL definitions. If you add the HCL functions in the `_test.go` files you'll clutter up them as well. Many `_test.go` files are already huge!

**Pull request merged**